### ISSUE TITLE
[Enhancement]: Single Product Page - Add To Cart Ajax

### ISSUE THREAD

**Issue by ABACHOPPA** [woothemes/woocommerce#36006]
### Describe the solution you'd like

Hello,

I would like to see an additional option under the Woocommerce Settings > Products > General > "Add to cart behaviour" section

To enable AJAX add to cart buttons on single product pages

![Screenshot 2022-12-15 at 9 11 46 am](https://user-images.githubusercontent.com/45242332/207819339-4fbe1263-057b-4784-bcd9-e6cbcf4be76f.png)

### Describe alternatives you've considered

I currently use a third party plugin, however this is adding to plugin bloat for a function that is already built into Woocommerce for the archives page and could be added to the single product pages also.

**Comment by barryhughes** [comment]
Agree that would be useful, and would add consistency.

**Comment by andrewklimek** [comment]
I got this to work by simply adding the classes `add_to_cart_button ajax_add_to_cart` and the `data-product_id` attribute to the single product button

**Comment by webaxones** [comment]
Hello,
Will this feature be coded, or is it too complex? 
The fact that it is missing blocks the use of FSE options such as "open when adding item" and even makes the use of the mini-cart-drawer block uninteresting since it operates asynchronously.
Thanks :)

**Comment by barryhughes** [comment]
> Will this feature be coded, or is it too complex?

I see no reason it wouldn't be implemented, and don't anticipate it being especially complex, but it's currently a low priority from our perspective.

**Comment by webaxones** [comment]
Hello @barryhughes,

From the perspective of freelancers and web agencies (who are contributing to the increasing use of WooCommerce), this is not a low priority but a high one. Let me explain: currently, many of us are contributing to the adoption of FSE by our e-commerce clients, whether on WordPress or WooCommerce. It's not a walk in the park. I've calculated that my workload has doubled due to the FSE. And I cannot pass on this additional cost to my clients. As with any new product, there are flaws, shortcomings, and bugs. We are participating in this adventure by reporting our issues and even offering solutions.

However, if you do not make our work easier, that's not nice either.

I already regretted that this feature was not available in 2016 when I developed the ecommerce site that I'm currently redesigning in FSE. Seven years later, I see that I can't use a major option of the FSE mini-cart-drawer block because the Core still hasn't integrated AJAX on the product page. It's really not nice of you.

**Comment by nerrad** [comment]
Low priority doesn't mean no priority :). We have to weigh any improvements against roadmap and our capacity. There are also a number of other factors to consider as we look at this:

- Why is the behaviour the way it is now and are there any back-compat considerations for changing. It's worth noting that this context (Single product page) is a bit different for "Add to Cart" because there could be different product types (attributes, variations, extensions adding customizations etc).
- Is this something to address for _all_ contexts, or is it better to invest the time in the features/behaviours provided by block themes/blocks? We have to weigh our time investments here.
- WordPress is [developing an Interactive API](https://make.wordpress.org/core/2023/03/30/proposal-the-interactivity-api-a-better-developer-experience-in-building-interactive-blocks/) which will standardize the approach to interactivity on the frontend. We have an initial implementation of this in the Woo Blocks repo and are planning on experimenting on that with the Add to Cart button initially. Learnings from that may apply to this context and we'd probably want to go with a future proof approach as opposed to implementing AJAX and changing it later.

With that said, this is on our radar and likely something we will address this year (especially in consideration of [this issue](https://github.com/woocommerce/woocommerce/issues/42638)).

**Comment by webaxones** [comment]
@nerrad I believe that there are 2 most common types of products: simple and variable. That AJAX for these 2 types of product is not complex to program if I base myself on Barry's response. And this would be a temporary solution that would allow us to hold out until the Interactive API. Because there's always a significant delay between the release of a technology and its adoption. Then, if I had to limit myself to one word, it's the one you said: consistency.
But above all, it would give us some breathing space in this tough task of adopting FSE on complex tools like WooCommerce :)

**Comment by tomxygen** [comment]
@nerrad do you have any ETA regarding the interactivity API and its implementation with the Add to cart button?
Currently, in order to have the add to cart + mini cart drawer that opens automatically I rely on a [third party plugin](https://wordpress.org/plugins/woo-ajax-add-to-cart/), which I would rather avoid in favor of an official implementation by WooCommerce itself.
I think this is a basic feature that is available on almost any ecommerce store.

**Comment by nerrad** [comment]
It's being [worked on now](https://github.com/woocommerce/woocommerce-blocks/pull/10006). There were some issues with [addressing caching](https://github.com/woocommerce/woocommerce-blocks/discussions/10153) that came up which is being discussed atm.

**Comment by tomxygen** [comment]
That's great! However I see that this issue is related to product Archives. Do you have any plans to introduce it for the single product page?

Also, what is the status of the Interactivity API? 
Is it ready for developers to use?

**Comment by nerrad** [comment]
> Also, what is the status of the Interactivity API?
> Is it ready for developers to use?

No, it's still a private API in Gutenberg. Woo is implementing a custom internal implementation of it so we can experiment with the architecture being implemented in WP and use that to provide feedback on gaps etc. We're working closely with the teams working on this in WP so that the API goes public with many of the usability gaps already addressed. This also ensures that we are able to adapt our implementation to what eventually surfaces in WP.

**Comment by tomxygen** [comment]
Hi @nerrad 
It's been a while and I haven't seen any updates regarding interactivity API on add to cart in the single product page.
Do you have any updates?

**Comment by nerrad** [comment]
@tomxygen - this is still something on the radar but our team capacity is being eaten up by other work atm. We'll get there.

**Comment by alnaggar-dev** [comment]
Any update guys ??

**Comment by Aljullu** [comment]
Hey there, @alnaggar-dev!

> Any update guys ??

We have this in our roadmap and will likely work on it as part of the project to blockify the Add to Cart with Options block (https://github.com/woocommerce/woocommerce/discussions/50484). We hadn't started it and we don't have an ETA, but we hadn't forgotten about it either. 🙂

**Comment by Aljullu** [comment]
Closing as this has been implemented in the _Add to Cart + Options_ block. 🙂
